ec2/instance.py

```diff
@@ -68,15 +78,19 @@ def run_instances(**_):
         return
 
     instance_parameters = _get_instance_parameters()
+    # Add in lifecycle input parameters if they exist
```

Comment:
all of this should be moved into _get_instance_parameters().
Add argument parameters to the function like
```python
def _get_instance_parameters(parameters):
    pass
````
probably before the _handle_userdata function.